roles/create-ai-environment/tasks/main.yaml

```diff
@@ -0,0 +1,289 @@
+- name: Set Default Hardware type Parameters
+  set_fact:
+    user_file: "user_data"
+  when: not is_gpu_enabled
+
+- name: Split Hardware type Parameters for GPU
+  set_fact:
+    user_file: "user_data_gpu"
+  when: is_gpu_enabled
+
+- name: Get User Metadata Zip or Url
+  shell:
+    cmd: |
+      set -x
+      set -e
+
+      apt install -y unzip
+      pip install pyyaml
+
+      result_folder="{{ ansible_user_dir }}/workspace/test_results/"
+      mkdir -p $result_folder
+
+      user_data_folder='{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/'
+
+      cd $user_data_folder
+
+      if ls | grep txt;then
+        txt_file=`ls | grep txt`
+        url=`cat $txt_file`
+        if [ -n "$url" ];then
+          wget $url
+        fi
+      fi
+
+      zip_file=`ls | grep zip`
+
+      unzip $zip_file
+      python '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/tools/analysis_yaml.py'
+    executable: /bin/bash
+
+- name: Get Runtime Info
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_runtime.txt'
+  register: runtime
+
+- name: Defind Parameters - runtime_name
+  shell: echo "{{ runtime.stdout | regex_findall('[java|go|python|c|c++]') | join("") }}"
+  register: runtime_name
+
+- name: Defind Parameters - runtime_version
+  shell: echo "{{ runtime.stdout | regex_replace('[java|go|python|c|c++]') }}"
+  register: runtime_version
+
+- name: Get Framework Name
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_name.txt'
+  register: framework
+
+- name: Get Framework Version
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_version.txt'
+  register: framework_version
+
+- name: Get User Entry Point
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/entry_point.txt'
+  register: entrypoint
+
+- name: Check output folder variable exists
```

Comment:
Output folder shouldn't be included in metadata.yaml, it's related with OpenLab, but not with AI framework and environment. And I prefer to use a fixed folder to save result for each job